<?xml version="1.0"?>
<launch>
    <!-- Parameters -->
    <arg name="use_sim_time" default="true"/>
    <arg name="robot_name" default="limo0"/>
    <arg name="use_gui" default="true"/>
    <arg name="use_rviz" default="true"/>

    <arg name="x_pos" default="1.0"/>
    <arg name="y_pos" default="0."/>
    <arg name="yaw_pos" default="0.0"/>

    <!--Spawn Gazebo-->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!--spawn the limo0 robot  -->
    <include file="$(find limo_description)/launch/upload.launch"/>
    <node name="spawn_model" pkg="gazebo_ros" type="spawn_model"
          args="-urdf -param robot_description -model $(arg robot_name) -x $(arg x_pos) -y $(arg y_pos)  -Y $(arg yaw_pos)"/>

    <!-- AMCL Localization -->
    <!--node pkg="amcl" type="amcl" name="amcl" output="screen"/-->

    <!-- Publish robot joint states -->
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher">
        <param name="rate" value="100.0"/>
    </node>

    <!-- Publish link TFs -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
        <param name="rate" value="100"/>
    </node>

    <!-- this emulates perfect no drift condition matching odom (that usually drifts) with map -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_limo1_odom"
          args="0 0 0 0 0 0 map  $(arg robot_name)/odom"/>
    
    <!-- RViz -->
    <!--node pkg="rviz" type="rviz" name="rviz" args="-d $(find loco_planning)/rviz/conf_tf.rviz"/-->

    <!-- keyboard -->
    <!--node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop_keyboard" output="screen"/-->

</launch>
